pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,268 @@
+	if dbPath == "" {
+		return nil, errors.New(blankDBPathErrMsg)
+	}
+
+	// Example DB Path root:my-secret-pw@tcp(127.0.0.1:3306)/
+	db, err := sql.Open("mysql", dbPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open connection: %w", err)
+	}
+
+	p := &Provider{
+		dbURL: dbPath,
+		db:    db,
+		dbs:   map[string]*sqlDBStore{}}
+
+	return p, nil
+}
+
+// OpenStore opens and returns new db for given name space.
+func (p *Provider) OpenStore(name string) (storage.Store, error) {
+	p.Lock()
+	defer p.Unlock()
+
+	// check name is not empty
+	if name == "" {
+		return nil, errors.New("store name is required")
+	}
+	// creating the database
+	_, err := p.db.Exec(createDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create db %s: %w", name, err)
+	}
+
+	// Opening new db connection
+	newDBConn, err := sql.Open("mysql", p.dbURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new connection %s: %w", p.dbURL, err)
+	}
+
+	// Use is used to select the created database without this DDL operations are not permitted
+	_, err = newDBConn.Exec(useDBQuery + name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to use db %s: %w", name, err)
+	}
+
+	tableName := tablePrefix + name
+	// key has max varchar size it can accommodate as per mysql 8.0 spec
+	createTableStmt := "CREATE Table IF NOT EXISTS " + tableName +
+		"(`key` varchar(255) NOT NULL ,`value` BLOB, PRIMARY KEY (`key`));"
+
+	// creating key-value table inside the database
+	_, err = newDBConn.Exec(createTableStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table %s: %w", tableName, err)
+	}
+
+	store := &sqlDBStore{
+		db:        newDBConn,
+		tableName: tableName}
+
+	p.dbs[name] = store
+
+	return store, nil
+}
+
+// Close closes the provider.
+func (p *Provider) Close() error {
+	p.Lock()
+	defer p.Unlock()
+
+	for _, store := range p.dbs {
+		err := store.db.Close()
+		if err != nil {
+			return fmt.Errorf(failToCloseProviderErrMsg+": %w", err)
+		}
+	}
+
+	if err := p.db.Close(); err != nil {
+		return err
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	return store.db.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" {
+		return errors.New("key is mandatory")
+	}
+
+	//nolint: gosec
+	// create upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	createStmt := "INSERT INTO " + s.tableName + " VALUES (?, ?) ON DUPLICATE KEY UPDATE value=?"
+	// executing the prepared insert statement
+	_, err := s.db.Exec(createStmt, k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to insert key and value record into %s %w ", s.tableName, err)
+	}
+
+	return nil
```

Comment:
Looks like there's a potential SQL injection attack here? Is that what the `gosec` warning was about?

Reply:
Mysql dont allow parameterized table name. But we required parametrized table name.  So thats why its false positive.

Reply:
I believe there's an SQL injection attack possibility here that needs to be addressed...unless this is somehow handled by the underlying library already. I found an article here that looks helpful: https://www.calhoun.io/what-is-sql-injection-and-how-do-i-avoid-it-in-go/

Reply:
^ It describes how to avoid injection attacks via escaping. Not sure if they're using the same library as you, but I assume there should be an equivalent if your library is different

Reply:
Yes escaping is what i am looking into , i have created a todo and will add that in the code, as i have to same thing to do for aries as well.

Reply:
> I believe there's an SQL injection attack possibility here that needs to be addressed...unless this is somehow handled by the underlying library already. I found an article here that looks helpful: https://www.calhoun.io/what-is-sql-injection-and-how-do-i-avoid-it-in-go/

I have looked into this, yes using the same library go-mysql, Escaping is the only via option for now.